File: FSharp/Hanoi/hanoi.fs
open System

let rec Hanoi n from to_ via =
  if n > 1 then
    Hanoi (n - 1) from via to_
    printf "%c -> %c\n" from to_
    Hanoi (n - 1) via to_ from
  else
    printf "%c -> %c\n" from to_

let n = Console.ReadLine() in
Hanoi (int n) 'a' 'b' 'c'

File: Swift/Hanoi/hanoi.swift
func hanoi(Num n: Int, From from: String, To to: String, Via via: String) {
    if (n > 1) {
        hanoi(Num: n - 1, From: from, To: via, Via: to)
        print(from + " -> " + to)
        hanoi(Num: n - 1, From: via, To: to, Via: from)
    } else {
        print(from + " -> " + to)
    }
}

let n = Int(readLine()!)!
hanoi(Num: Int(n), From: "a", To: "b", Via: "c")
